Library/Source/ShaderCompilerOpenGL.cpp

```diff
@@ -1,19 +1,164 @@
 #include "ShaderCompiler.h"
+#include "ResourceLimits.h"
+#include <arcana/experimental/array.h>
+#include <glslang/Public/ShaderLang.h>
+#include <SPIRV/GlslangToSpv.h>
+#include <spirv_parser.hpp>
+#include <spirv_glsl.hpp>
 
 namespace Babylon
 {
+    extern const TBuiltInResource DefaultTBuiltInResource;
+
+    namespace
+    {
+        void AddShader(glslang::TProgram& program, glslang::TShader& shader, std::string_view source)
+        {
+            const std::array<const char*, 1> sources{ source.data() };
+            shader.setStrings(sources.data(), gsl::narrow_cast<int>(sources.size()));
+            shader.setEnvInput(glslang::EShSourceGlsl, shader.getStage(), glslang::EShClientVulkan, 100);
+            shader.setEnvClient(glslang::EShClientVulkan, glslang::EShTargetVulkan_1_0);
+            shader.setEnvTarget(glslang::EShTargetSpv, glslang::EShTargetSpv_1_0);
+
+            if (!shader.parse(&DefaultTBuiltInResource, 450, false, EShMsgDefault))
+            {
+                throw std::exception();
+            }
+
+            program.addShader(&shader);
+        }
+
+        std::unique_ptr<spirv_cross::Compiler> CompileShader(glslang::TProgram& program, EShLanguage stage, std::string& glsl)
+        {
+            std::vector<uint32_t> spirv;
+            glslang::GlslangToSpv(*program.getIntermediate(stage), spirv);
+
+            spirv_cross::Parser parser{ std::move(spirv) };
+            parser.parse();
+
+            auto compiler = std::make_unique<spirv_cross::CompilerGLSL>(parser.get_parsed_ir());
+            
+            compiler->build_combined_image_samplers();
+            
+            spirv_cross::CompilerGLSL::Options options = compiler->get_common_options();
+
+#ifdef ANDROID
+            options.version = 300;
```

Comment:
Can we specify these values via compiler defines or something, perhaps in CMake? Not sure if that would be better, but they seem like magic numbers here.

Reply:
Not that magic, it's asking for #version 300 es